src/indexmap.rs

```diff
@@ -0,0 +1,2539 @@
+        }
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn iter_doctest() {
+        let mut map = FnvIndexMap::<_, _, 16>::new();
+        map.insert("a", 1).unwrap();
+        map.insert("b", 2).unwrap();
+        map.insert("c", 3).unwrap();
+        for (key, val) in map.iter() {
+            println!("key: {} val: {}", key, val);
+        }
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn iter_mut_doctest() {
+        let mut map = FnvIndexMap::<_, _, 16>::new();
+        map.insert("a", 1).unwrap();
+        map.insert("b", 2).unwrap();
+        map.insert("c", 3).unwrap();
+        for (_, val) in map.iter_mut() {
+            *val = 2;
+        }
+        for (key, val) in &map {
+            println!("key: {} val: {}", key, val);
+        }
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn entry_doctest() {
+        let mut map = FnvIndexMap::<_, _, 16>::new();
+        if let Entry::Vacant(v) = map.entry("a") {
+            v.insert(1).unwrap();
+        }
+        if let Entry::Occupied(mut o) = map.entry("a") {
+            println!("found {}", *o.get()); // Prints 1
+            o.insert(2);
+        }
+        // Prints 2
+        println!("val: {}", *map.get("a").unwrap());
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn len_doctest() {
+        let mut a = FnvIndexMap::<_, _, 16>::new();
+        assert_eq!(a.len(), 0);
+        a.insert(1, "a").unwrap();
+        assert_eq!(a.len(), 1);
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn is_empty_doctest() {
+        let mut a = FnvIndexMap::<_, _, 16>::new();
+        assert!(a.is_empty());
+        a.insert(1, "a");
+        assert!(!a.is_empty());
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn clear_doctest() {
+        let mut a = FnvIndexMap::<_, _, 16>::new();
+        a.insert(1, "a");
+        a.clear();
+        assert!(a.is_empty());
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn get_doctest() {
+        let mut map = FnvIndexMap::<_, _, 16>::new();
+        map.insert(1, "a").unwrap();
+        assert_eq!(map.get(&1), Some(&"a"));
+        assert_eq!(map.get(&2), None);
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn contains_key_doctest() {
+        let mut map = FnvIndexMap::<_, _, 8>::new();
+        map.insert(1, "a").unwrap();
+        assert_eq!(map.contains_key(&1), true);
+        assert_eq!(map.contains_key(&2), false);
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn get_mut_doctest() {
+        let mut map = FnvIndexMap::<_, _, 8>::new();
+        map.insert(1, "a").unwrap();
+        if let Some(x) = map.get_mut(&1) {
+            *x = "b";
+        }
+        assert_eq!(map[&1], "b");
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn insert_doctest() {
+        let mut map = FnvIndexMap::<_, _, 8>::new();
+        assert_eq!(map.insert(37, "a"), Ok(None));
+        assert_eq!(map.is_empty(), false);
+        map.insert(37, "b");
```

Comment:
```suggestion
        let _ = map.insert(37, "b").unwrap();
```

This should never fail, so unwrapping in the test is good.

Reply:
Why is `let _ =` necessary?

Reply:
I'm guessing it is to please clippy / and uphold more correct code convention? 

Since the method returns a value, you should always be assigning it to something. But since you don't care about using the returned value, you should be using underscore (discard) to illustrate this.

Correct me if I'm wrong here @alice-i-cecile

Reply:
Actually, I think Cole is right! Because we're calling `unwrap()` on this the intent is clear + Clippy is satisfied.

I'd initially gone with just the `let _` solution, but then added the unwrap. They're redundant, and unwrap should be preferred.

Reply:
```suggestion
        map.insert(37, "b").unwrap();
```